### PR TITLE
dialog/focusmanager: update to react16 and hooks

### PR DESCRIPTION
**Pull request by danethurber**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -43,3 +43,6 @@ jspm_packages
 .env*
 
 .cache
+
+# workspace config
+.vim
```
Comment (jaketrent) [inline .gitignore]: I would prefer personal dev environment ignores to happen at a system level, outside the project.  See  https://jaketrent.com/post/gitignore-system-files/

Does that seem reasonable?

File: packages/dialog/src/react/index.js [hunk 2]
```diff
@@ -1,174 +1,151 @@
...
+  children: PropTypes.node,
   disableCloseButton: PropTypes.bool,
   disableCloseOnEscape: PropTypes.bool,
+  disableCloseOnOverlayClick: PropTypes.bool,
   disableFocusOnMount: PropTypes.bool,
   modal: PropTypes.bool,
   onClose: PropTypes.func,
-  style: PropTypes.object,
-  tailPosition: PropTypes.oneOf(Object.keys(vars.tailPositions))
+  tailPosition: PropTypes.oneOf(Object.keys(Dialog.tailPositions))
```
Comment (jaketrent) [inline packages/dialog/src/react/index.js]: You want values instead of keys, right?

File: .gitignore (hunk 1 above)
Comment (danethurber) [reply .gitignore]: good idea. i'll move it

File: packages/dialog/src/react/index.js (hunk 2 above)
Comment (danethurber) [reply packages/dialog/src/react/index.js]: yes....will update

**Pull request merged**